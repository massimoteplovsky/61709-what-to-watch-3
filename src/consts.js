export const DEFAULT_GENRE = `All genres`;
export const FILM_TO_SHOW = 8;
export const RELATED_FILM_COUNT = 4;
export const AUTH = `AUTH`;
export const NO_AUTH = `NO_AUTH`;
export const BASE_URL = `https://htmlacademy-react-2.appspot.com`;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const NOT_FAVORITE_FILM = 0;
export const FAVORITE_FILM = 1;
export const MAX_RATING = 5;
