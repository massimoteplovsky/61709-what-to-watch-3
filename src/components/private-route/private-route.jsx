import React from "react";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AUTH} from "../../consts.js";
import {getAuthorizationStatus} from "../../selectors/user/user.js";
import {PropTypes} from "prop-types";

const PrivateRoute = ({authorizationStatus, path, exact, computedMatch, render}) => {
  const {id} = computedMatch.params;
  return (
    <Route
      path={path}
      exact={exact}
      render={() => {
        return (
          authorizationStatus === AUTH ?
            render(id)
            :
            <Redirect to="/login"/>
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  computedMatch: PropTypes.object
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
