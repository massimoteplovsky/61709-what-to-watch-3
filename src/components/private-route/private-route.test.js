import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route.jsx";
import {AUTH} from "../../consts.js";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../history.js";

const mockStore = configureMockStore([]);
let store = mockStore({
  user: {
    authorizationStatus: AUTH
  }
});

it(`<PrivateRoute /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              authorizationStatus={AUTH}
              exact={true}
              path={`/`}
              render={() => {}}
              computedMatch={{params: {id: 1}}}
            />
          </Router>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});
