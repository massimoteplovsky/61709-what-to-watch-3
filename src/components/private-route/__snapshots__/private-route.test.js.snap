// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> component renders correctly 1`] = `null`;
